### ISSUE TITLE
Allow disable `bulkActionButtons` in Datagrid while still allowing selectable rows

### ISSUE THREAD

**Issue by andrico1234** [marmelab/react-admin#7677]
With the `useSelectContext` hook, it gives us the opportunity to extend our datagrids without having to reimplement a lot of UI from scratch, which is amazing. 

One example is a custom cancel button like the following:

<img width="752" alt="image" src="https://user-images.githubusercontent.com/17087167/167612071-0c896b3f-828e-47ca-a5d4-b5305098d1d4.png">

We have a cool sticky confirm button, which renders the default `bulkActionButtons` redundant. It would be great to have an api like the following that would allow us to hide the `bulkActionButtons` while also ensuring all rows are selectable. 

My first guess would be the following:

```
<Datagrid
  // This renders a span where the buttons themselves are supposed to be
  // But causes the overall component to still display
  bulkActionButtons={<span />}
  // or

  // This hides the action buttons, but this takes precedence over the `isRowSelectable` property
  bulkActionButtons={false}
  isRowSelectable
></Datagrid>;
        
```

Since this is not a primary use case, I wouldn't expect any specific API to allow this, but I would say that the second option would make sense. `isRowSelectable` seems more specific than the broader `bulkActionButtons` property. Having `isRowSelectable` set to `true` is the developer explictly stating that they want the select boxes to display.

I'll be happy to open a PR if you're happy with my suggestion

**Comment by slax57** [comment]
Hi!
If I understand correctly, you did not want to add your "Cancel Appointments" button inside the `bulkActionButtons` because you want to have a custom positioning, is that correct?

Can't you try customizing the style of the `BulkActionsToolbar` component rather?

Personally, I like the simplicity of the `bulkActionButtons` pattern. Also, the `isRowSelectable` is here to provide a way to enable the checkboxes on a per-row basis, but not on the whole Datagrid.
TLDR; I personally would not encourage this feature.
What about you @djhi  and @fzaninotto ?

**Comment by andrico1234** [comment]
Ahh, I hadn't even considered hiding the BulkActionsToolbar. I agree, if that's something that we can do then there\s absolutely no need to extend the API

**Comment by fzaninotto** [comment]
Agreed, this is something that can be done using a custom BulkActionsToolbar. We won't work on an alternative solution.
